srcCxx/ShapeType_main.cxx

```diff
@@ -54,8 +65,50 @@ void setup_signal_handler()
     signal(SIGABRT, signal_handler); //Abort
 }
 
+/*************************************************************/
+enum Verbosity
+{
+    ERROR=1,
+    DEBUG=2,
+};
+
+class Logger{
+public:
+    Logger(enum Verbosity v)
+    {
+        verbosity_ = v;
+    }
+
+    void verbosity(enum Verbosity v)
+    {
+        verbosity_ = v;
+    }
+
+    enum Verbosity verbosity()
+    {
+        return verbosity_;
+    }
+
+    void log_message(std::string message, enum Verbosity level_verbosity)
+    {
+        if (level_verbosity <= verbosity_) {
+            std::cout << message << std::endl;
+        }
+    }
+
+private:
+    enum Verbosity verbosity_;
+};
+
+/*************************************************************/
+Logger logger(ERROR);
+
+
+/*************************************************************/
 Topic *create_topic(DomainParticipant *participant, const char *topic_name)
 {
+    printf("Create topic: %s\n", topic_name );
```

Comment:
I see that the `dds-rtps` project is also using the logger mostly for errors, and mixing it with `printf`. Maybe in the future we could add other verbosities and replace the current calls to `printf` with `logger.log_message`.

Reply:
I agree, I struggled with the decision of how far to go with this change...
I ended up just trying to keep it similar to dds-rtps.
I do think it makes sense to transition all printf's into logger calls.  Maybe it would be good to write a new [issue](https://github.com/omg-dds/dds-security/issues) to go through the whole code base and do that?